### PR TITLE
Fix perf issue on get locations of breakpoint

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>
fix bug #49.

Nested types are just needed for those already loaded classes. For the referenceType from ClassPrepareEvent, no need to search the locations in the nested types. For newly loaded classes, the debugger has also registered ClassPrepareRequest for their local types, so the debugger will handle the local types in ClassPrepareEvent handler.

### CODE REVIEW

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java [hunk 1]
```diff
@@ -167,30 +168,44 @@ public void setHitCount(int hitCount) {
...
+    private static List<Location> collectLocations(List<ReferenceType> refTypes, int lineNumber, boolean includeNestedTypes) {
         List<Location> locations = new ArrayList<>();
         try {
             refTypes.forEach(refType -> {
-                locations.addAll(collectLocations(refType, lineNumber));
-                locations.addAll(collectLocations(refType.nestedTypes(), lineNumber));
+                List<Location> newLocations = collectLocations(refType, lineNumber);
+                if (!newLocations.isEmpty()) {
+                    locations.addAll(newLocations);
+                } else if (includeNestedTypes) {
```
Comment (ansyral) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java]: just if?

Comment (testforstephen) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java]: what's the advice?

Comment (ansyral) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java]: i mean why here it is `else if` rather than `if`?

Comment (testforstephen) [reply com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java]: if the breakpoint located in the refType, no need to search the nested type.

**Comment by Goom11** [comment]
Hello, @testforstephen, how do you know that ReferenceType.nestedTypes invokes VirtualMachine.allClasses?

**Pull request merged**